documentation/milestones/individual_milestones/get_opportunities.md

```diff
@@ -0,0 +1,196 @@
+  - changes made to a given endpoint to be backward-compatible, so that I can start building against this API without worrying about breaking changes
+- As a **project maintainer**, I want:
+  - most of the critical development infrastructure to be in place when we officially launch the API, so that we can deploy bug fixes or new features quickly once the site is live
+- As an **open source contributor**, I want:
+  - to be able run the code for the API locally, so that I can test my contributions to the codebase
+
+## Technical description
+
+### Infrastructural Requirements
+
+The infrastructure developed to deploy and host the API should balance:
+
+- Code quality
+- Security
+- Delivery velocity
+- Cost & maintenance
+
+### Developer Experience Requirements
+
+The API should be developed and versioned in a way that balances:
+
+- Discoverability
+- Ease of adoption
+- Backwards compatibility
+
+### Data Requirements
+
+The way that data is stored and delivered through the API should balance:
+
+- Eventual consistency with legacy Grants.gov
+- Improvements to the existing data model
+- Ease of managing schema changes
+
+### Definition of done
+<!-- Required -->
+
+- [ ] The following infrastructure requirements are satisfied:
+  - [ ] The code needed to build and deploy the site is merged to `main`
+  - [ ] The site is built and hosted with the tools selected in the [API Planning](https://github.com/HHS/grants-equity/issues/42) and [DB Planning](https://github.com/HHS/grants-equity/issues/48) milestones
+  - [ ] All code quality checks set up in the [Developer Tools milestone](https://github.com/HHS/grants-equity/issues/50) are passing
+  - [ ] The resources required to deploy and host the API are provisioned programmatically using the [Infrastructure-as-Code milestone](https://github.com/HHS/grants-equity/issues/123) framework
+  - [ ] Code changes are deployed using the CI/CD pipeline set up in [the Back-end CI/CD milestone](https://github.com/HHS/grants-equity/issues/57)
+  - [ ] Logging/monitoring is configured, and it both records the metrics defined below and alerts the development team when the API is down
+- [ ] The following developer experience (DX) requirements are satisfied:
+  - [ ] The API is live at `beta.grants.gov/api/`
+  - [ ] Developers can learn how to interact with the API by referencing the API documentation
+  - [ ] The endpoint path indicates which major version of the API the developer is consuming
+  - [ ] Breaking changes are reflected in predictable changes to the endpoint path
+  - [ ] The endpoint is available when legacy grants.gov experiences planned or unplanned downtime, maintenance, and upgrades
+- [ ] The following data requirements are satisfied:
+  - [ ] The endpoint returns all of the grant opportunities that are available on grants.gov
+  - [ ] The endpoint returns at least one (1) field per opportunity
+  - [ ] Updates to the data in legacy Grants.gov are propagated to the new endpoint within 24 hours
```

Comment:
Hmm. 24 hours actually feels like a long window. Do we think we can have a technical approach that gets it to, say, 1 hour or 1 minute? 

We really don't want users to be downloading an out-of-date NOFO, or missing a NOFO opportunity that went live on g.gov because they're browsing on beta.grants.gov. 

With an AWS replica DB, it should be instantly available, but perhaps we're planning some batch processing steps that convert data from the existing format in the replica DB to the new format.

Reply:
Instantaneous might be tough if we expect to migrate the data to a new data model, which will probably need to run on some kind of a batch schedule. I did reduce this to 1 hour though in [this commit](https://github.com/HHS/grants-equity/pull/249/commits/3aa14fa418605012f4e9766952271ca22b97e8f2)